rails/app/controllers/portal/offerings_controller.rb

```diff
@@ -58,8 +58,9 @@ def show
            # session[:put_path] = nil
          end
          external_activity = @offering.runnable
-         if external_activity.launch_url.present?
-           uri = URI.parse(external_activity.launch_url)
+         tool = Tool.where(id: external_activity.tool_id).first
```

Comment:
```suggestion
         tool = external_activity.tool
```

Reply:
Thanks, @scytacki. I did this change a little differently. Does what I did make sense? It seems to work fine.